src/NuGet.Services.EndToEnd/Support/Clients/V2V3SearchClient.cs

```diff
@@ -0,0 +1,253 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.WebUtilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NuGet.Services.EndToEnd.Support
+{
+    public class V2V3SearchClient
+    {
+        private readonly V3IndexClient _v3IndexClient;
+        private readonly SimpleHttpClient _httpClient;
+        private readonly TestSettings _testSettings;
+
+        public V2V3SearchClient(SimpleHttpClient httpClient, V3IndexClient v3IndexClient, TestSettings testSettings)
+        {
+            _httpClient = httpClient;
+            _v3IndexClient = v3IndexClient;
+            _testSettings = testSettings;
+        }
+
+        public async Task<V2SearchResponse> SearchQuery(string queryString)
+        {
+            var searchBaseUrls = new List<string>();
+            if (_testSettings.SearchBaseUrl != null)
+            {
+                searchBaseUrls.Add(_testSettings.SearchBaseUrl);
+            }
+            else
+            {
+                searchBaseUrls.AddRange(await _v3IndexClient.GetSearchBaseUrls());
+            }
+
+            var queryUrl = searchBaseUrls
```

Comment:
Hm... `searchBaseUrl` should be a parameter. All search services should be tested.